pkg/volmount/getvolumenameforvolumemountpoint.go

```diff
@@ -0,0 +1,33 @@
+package volmount
+
+import (
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// GetVolumeNameForVolumeMountPoint returns a volume path (in format '\\?\Volume{GUID}'
+// for the volume mounted at targetPath.
+// https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumenameforvolumemountpointw
+func GetVolumeNameForVolumeMountPoint(targetPath string) (string, error) {
+	// Must end in a backslash
+	slashedTarget := filepath.Clean(targetPath)
+	if slashedTarget[len(slashedTarget)-1] != filepath.Separator {
+		slashedTarget = slashedTarget + string(filepath.Separator)
+	}
+
+	targetP, err := windows.UTF16PtrFromString(slashedTarget)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to utf16-ise %s", slashedTarget)
+	}
+
+	bufferlength := uint32(50) // "A reasonable size for the buffer" per the documentation.
+	buffer := make([]uint16, bufferlength)
+
+	if err = windows.GetVolumeNameForVolumeMountPoint(targetP, &buffer[0], bufferlength); err != nil {
+		return "", errors.Wrapf(err, "failed calling GetVolumeNameForVolumeMountPoint('%s', ..., %d)", slashedTarget, bufferlength)
```

Comment:
Personally I feel that rewriting the Win32 API name into prose doesn't gain much clarity, and loses the specific detail that it was a syscall that failed. But I might be missing your point here, so happy to hear more.

Reply:
My only concern was that since we are wrapping win32 APIs our error messages can also be less specific to the API and avoid sharing the internal details with the caller. But I agree that knowing that a particular syscall failed can be useful so I am fine either way. For now we can keep it as it is and change it later if required.